MediaRecorder.html

```diff
@@ -442,6 +443,46 @@
               <em>Return type:</em> <code>boolean</code>
             </div>
           </dd>
+
+          <dt><dfn><code>getSupportedTypes</code></dfn>, static</dt>
+          <dd>
+            When this method is invoked, the User Agent must run the following
+            steps:
+            <ol>
+            <li>Let <var>p</var> be a promise.</li>
+            <li>Run the following steps in parallel:
+             <ol>
+             <li>Let <var>resultList</var> be an empty list.
+             <li>For each <a href="https://tools.ietf.org/html/rfc2046">MIME
+             type</a> <var>type</var> that is supported for encoding and/or
+             multiplexing (i.e. <code>isSuportedType(<var>type</var>)</code>
+             is true), add <var>type</var> to <var>resultList</var>.
+
+             <var>type</var> MUST be a <a def-id="mime-types">valid MIME type
+             </a> [[!HTML5]]. The codecs parameter [[!RFC6381]], if necessary to
+             specify exactly how the resource is encoded, SHOULD be added
+             to <code>type</code>.
+             If <var>type</var> can be expressed in several equivalent ways
+             (e.g. with or without quotes and whitespace, or different
+             capitalization), the shortest form with the largest number of
+             lowercase characters MUST be used.
+
+             </a>.
+             <var>resultList</var> MAY be ordered by the UA so that MIME types
+             closer to the beginning are preferred, for example due to larger
```

Comment:
This doesn't allow the application to distinguish which are actually encoded or otherwise preferred. For example, if there are three results, all, two, one, or none of them might be accelerated. Applications may end up just picking the first result.

Perhaps this should use the same mechanism as Media Capabilities ends up using. See the [discussion](https://github.com/WICG/media-capabilities/blob/master/explainer.md#decoding-capabilities) of smoothness and power efficiency in the explainer.